lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -2477,6 +2481,20 @@ submitTransaction ctx apiw@(ApiT wid) apitx@(ApiSerialisedTransaction (ApiT seal
             all isInpForeign generalInps &&
             all isWdrlForeign generalWdrls
 
+    sameInput inp1 inp2 = case (inp1, inp2) of
+        (WalletInput (ApiWalletInput _ _ _ derPath1 _ _), WalletInput (ApiWalletInput _ _ _ derPath2 _ _) ) ->
+               derPath1 == derPath2
+        (ExternalInput txin1, ExternalInput txin2) ->
```

Comment:
for external inputs we really cannot do better than that. we don't have knowledge of derivation paths. So in essence, we are not exact here. We provide reasonable upper bound. I will think about it when adding support for partial signing though